teos/src/cli.rs

```diff
@@ -75,11 +75,21 @@ async fn main() {
             println!("{}", pretty_json(&appointments.into_inner()).unwrap());
         }
         Command::GetAppointments(appointments_data) => {
+            let mut user_id = None;
+            if let Some(i) = &appointments_data.user_id {
+                match UserId::from_str(i.as_str()) {
+                    Ok(parsed_user_id) => {
+                        user_id = Some(parsed_user_id.to_vec());
+                    }
+                    Err(err) => handle_error(err),
+                }
+            }
```

Comment:
simplify to:
```rust
            let user_id = if let Some(id) = &appointments_data.user_id {
                UserId::from_str(id.as_str())
                    .map_err(|err| handle_error(err))
                    .ok()
            } else {
                None
            };
```

Reply:
You can also nest the whole thing using `transpose`:

```suggestion
            match appointments_data
                .user_id
                .map(|id| UserId::from_str(id.as_str()).map(|id| id.to_vec()))
                .transpose()
            {
                Ok(user_id) => {
                    match Locator::from_hex(&appointments_data.locator) {
                        Ok(locator) => {
                            match client
                                .get_appointments(Request::new(msgs::GetAppointmentsRequest {
                                    locator: locator.to_vec(),
                                    user_id,
                                }))
                                .await
                            {
                                Ok(appointments) => {
                                    println!("{}", pretty_json(&appointments.into_inner()).unwrap())
                                }
                                Err(status) => handle_error(status.message()),
                            }
                        }
                        Err(e) => handle_error(e),
                    };
                }
```

That way all the error logic gets pushed to the bottom, even though there is more nesting